From f98d9c88b8af3c24df80ee23ada46dc9723f94aa Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Wed, 3 May 2023 11:27:56 +0530
Subject: [PATCH 57/65] usb: dwc3: Fix for usb3 configuration with no usb3-phy

Missing property may also be signaled via -EINVAL
Fixes: 868d58f69c7c ("usb: dwc3: Fix non-usb3 configurations")
Fixes: 142d50fbce7c ("usb: dwc3: Add support for usb3-phy PHY configuration")

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git e0061a2c78ea3c78ee40867002cd5c851fd4eacf usb: dwc3: Fix for usb3 configuration with no usb3-phy]
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Change-Id: I4db39d0d853d6bfcf3bae9b0bb4b7ed181515ca4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/303905
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 drivers/usb/dwc3/dwc3-generic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/dwc3-generic.c b/drivers/usb/dwc3/dwc3-generic.c
index 13d1d8ca7e9..2cb1596bbc3 100644
--- a/drivers/usb/dwc3/dwc3-generic.c
+++ b/drivers/usb/dwc3/dwc3-generic.c
@@ -691,7 +691,7 @@ int dwc3_glue_probe(struct udevice *dev)
 		ret = generic_phy_init(&phy);
 		if (ret)
 			return ret;
-	} else if (ret != -ENOENT && ret != -ENODATA) {
+	} else if (ret != -ENOENT && ret != -ENODATA && ret != -EINVAL) {
 		debug("could not get phy (err %d)\n", ret);
 		return ret;
 	}
-- 
2.51.0

